internal/services/ch/queries.go

```diff
@@ -343,3 +343,15 @@ func getFilterMods(filter *model.SignalFilter) []qm.QueryMod {
 	}
 	return mods
 }
+
+// getDeviceActivityQuery returns the most recent 3 hour bucket during which device activity was recorded.
+func getDeviceActivityQuery(bucketInterval int64, vehicleTokenID int, adManuf string) (string, []any) {
+	mods := []qm.QueryMod{
+		qm.Select(fmt.Sprintf("toStartOfInterval(max(%s), toIntervalMillisecond(%d)) as %s", vss.TimestampCol, bucketInterval*time.Hour.Milliseconds(), IntervalGroup)),
```

Comment:
I would prefer doing the smearing outside of SQL. I don't think we gain much here.